lib/electric_slide/agent.rb

```diff
@@ -1,6 +1,6 @@
 # encoding: utf-8
 class ElectricSlide::Agent
-  attr_accessor :id, :address, :presence, :call, :connect_callback, :disconnect_callback
+  attr_accessor :id, :address, :presence, :call, :connect_callback, :disconnect_callback, :agent_available_callback, :connection_failed_callback
```

Comment:
Rather than have a specific callback for the available state, I think it will make more sense to have a generic state-change event, which will include the new state as data passed to it.